src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1347,6 +1347,11 @@ protected override void ProcessRecord()
             {
                 foreach (string filename in expandedPaths)
                 {
+                    if (Directory.Exists(filename))
```

Comment:
Hmm... I actually think we should only ignore directories if they're among the paths that result from _wildcard_ resolution.

If you specify an effectively literal directory path - whether via `-LiteralPath` or via `-Path` and a path that happens not to contain (unescaped) wildcard characters - you should continue to get an error (though arguably a better one than currently).

This will let users know that specifying directories is not supported. With this fix as-is, If you do something like `Select-String foo .`, you'll get no output and no error, which could be misinterpreted as the string not having been found in any files in the directory.

I think the behavior should be analogous to the following `Get-ChildItem` behavior:

```powershell
Get-ChildItem -Directory -LiteralPath /NoSuchDir  # error
Get-ChildItem -Directory -Path /NoSuchDir  # error too, because the path contains no (unescaped) wildcard chars.
Get-ChildItem -Directory -Path /NoSuchDir*  # NO error, because the wildcard expression matched nothing.
```

Reply:
Good point. We can reactivate #4820 and get the behavior fixed. @iSazonov what do you think?

Reply:
Now `Select-String` behavior is the same as `Unblock-File`. There may be other examples (?)
So I believe it is _another issue_.
I agree that we could fix this with a "breaking change" symptom. If @mklement0 would done the analysis (what cmdlets should we fix?), I'd have solved the Issue.